### PR TITLE
#2075 Update naming conventions

### PR DESCRIPTION
**Pull request by cossartque**

### CODE REVIEW

File: dynawo/sources/Models/Modelica/Dynawo/UsersGuide/NamingConventions.mo [hunk 1]
```diff
@@ -55,16 +55,14 @@ annotation(preferredView = "info",
 <strong>Functions</strong>: start with lower case letter</li>
 </ul>
 
-<p>In addition to this, please follow the additional naming conventions bellow:</p>
+<p>In addition to this, please follow the additional naming conventions bellow (in that order):</p>
```
Comment (ErwanGuichard49) [inline dynawo/sources/Models/Modelica/Dynawo/UsersGuide/NamingConventions.mo:58]: below, not bellow

File: dynawo/sources/Models/Modelica/Dynawo/UsersGuide/NamingConventions.mo [hunk 2]
```diff
@@ -55,16 +55,14 @@ annotation(preferredView = "info",
...
+<li><strong>Min/Max</strong>&nbsp;parameters/variables: end with&nbsp;<code>Min/Max</code></li><li>
+<strong>Start</strong> parameters/variables: end with <code>0</code></li>
 <li>
-<strong>Start</strong> parameters/variables: end with <code>Start</code></li>
-<li>
-<strong>Per-unit</strong> parameters/variables: end with <code>Pu</code></li>
-<li>
-<strong>Min/Max</strong> parameters/variables: end with <code>Min/Max</code></li>
-</ul>
+<strong>Per-unit</strong> parameters/variables: end with <code>Pu (always indicate \"in p.u\" in the comment along with the p.u base)</code></li>
```
Comment (ErwanGuichard49) [inline dynawo/sources/Models/Modelica/Dynawo/UsersGuide/NamingConventions.mo:64]: According to Wikipedia, the abbreviation of per-unit is pu without a dot.

File: dynawo/sources/Models/Modelica/Dynawo/UsersGuide/NamingConventions.mo [hunk 3]
```diff
@@ -116,18 +112,17 @@ e.g tInteger, t0...</li>
 <li>Other libraries imports (alphabetical order)</li>
 <li>Internal Dynawo imports (alphabetical order)</li>
 <li>extends statements</li>
-<li>public items: parameters, connectors (group them based on functional aspects)</li>
-<li>protected items: parameters, variables, blocks (group them based on functional aspects)</li>
-<li>equations: complex and intermediate variables management equations first, followed by when equations and if equations</li>
-<li>annotations: remove minor tool-specific annotations</li>
+<li>Public parameters (including extends)</li><li>Connectors (group them based on functional aspects), inputs/outputs (in that order)</li><li>Blocks&nbsp;(group them based on functional aspects)</li><li>Initial parameters (if not protected)</li><li>Protected parameters and protected final parameters</li><li>Protected initial parameters and protected initial final parameters</li><li>Protected variables</li>
```
[the remark this answers is not in the document]
Comment (ErwanGuichard49) [reply dynawo/sources/Models/Modelica/Dynawo/UsersGuide/NamingConventions.mo:115]: Could the public and protected elements be written on separate lines ?

File: dynawo/sources/Models/Modelica/Dynawo/UsersGuide/NamingConventions.mo (hunk 2 above)
Comment (cossartque) [reply dynawo/sources/Models/Modelica/Dynawo/UsersGuide/NamingConventions.mo:64]: Let's keep p.u for the moment as it is the most used in the library. We could do a review later on when we have time.

File: dynawo/sources/Models/Modelica/Dynawo/UsersGuide/NamingConventions.mo [hunk 4]
```diff
@@ -165,6 +160,6 @@ e.g tInteger, t0...</li>
...
...li><b>No</b> blank line between the model name and the first import</li><li>One blank line between the imports and the extends</li><li>One blank line between the extends and the parameters</li><li>One blank line between the parameters and the connectors/inputs/outputs</li><li>One blank line between the connectors/input/outputs and the blocks</li><li>One blank line between the blocks and the initial parameters</li><li>One blank line between the initial parameters and the protected section</li><li><b>No</b> blank line between the \"protected\" word and the first protected element</li><li>One blank line between the protected parameters and the protected init parameters</li><li>One blank line between the protected init parameters and protected variables</li><li>One blank line between the protected variables and the equation section</li><li><b>No</b> blank line between the \"equation\" word and the first equation</li><li>One blank line between equations and a when equation or an if equation</li><li>One blank line before annotations</li><li><b>No</b> blank line between annotations and end of model</li></ul><div>Two spaces must be used for indentation.</div></div><p></p></body></html>"));
```
Comment (ErwanGuichard49) [inline dynawo/sources/Models/Modelica/Dynawo/UsersGuide/NamingConventions.mo:163]: I would prefer a blank line between the model name and the first import line, but I'll do as you wish.

File: dynawo/sources/Models/Modelica/Dynawo/UsersGuide/NamingConventions.mo (hunk 1 above)
Comment (cossartque) [reply dynawo/sources/Models/Modelica/Dynawo/UsersGuide/NamingConventions.mo:58]: Ok, thanks

File: dynawo/sources/Models/Modelica/Dynawo/UsersGuide/NamingConventions.mo (hunk 4 above)
Comment (cossartque) [reply dynawo/sources/Models/Modelica/Dynawo/UsersGuide/NamingConventions.mo:163]: To be consistent with the rest of the library and what we do with protected and equation sections

**Review by adriguir — APPROVED** [review]

**Comment by adriguir** [comment]
It now needs to be enforced throughout the whole library ;)

**Comment by cossartque** [comment]
> It now needs to be enforced throughout the whole library ;)

A very interesting job indeed!

**Pull request merged**